src/finch/codegen/c.py

```diff
@@ -516,17 +528,41 @@ def block(self) -> "CContext":
         blk.headers = self.headers
         blk._headerset = self._headerset
         blk.bindings = self.bindings
+        blk.slots = self.slots
+        blk.fptr = self.fptr
         return blk
 
     def subblock(self):
         blk = self.block()
         blk.indent = self.indent + 1
         blk.bindings = self.bindings.scope()
+        blk.slots = self.slots.scope()
         return blk
 
+    def resolve(self, node):
+        match node:
+            case asm.Slot(var_n, var_t):
+                if var_n in self.slots:
+                    var_o = self.slots[var_n]
+                    return asm.Stack(var_o, var_t)
+                raise KeyError(f"Slot {var_n} not found in context")
+            case asm.Stack(_, _):
+                return node
+            case _:
+                raise ValueError(f"Expected Slot or Stack, got: {type(node)}")
+
     def emit(self):
         return "\n".join([*self.preamble, *self.epilogue])
 
+    def cache(self, name, val):
```

Comment:
Is `cache` still needed? It's used in only one place.

Reply:
I'd like to keep it around, it comes in handy sometimes.